### PR TITLE
Fix regression caused by 26620

### PR DESCRIPTION
**Pull request by mattwire**
Overview
----------------------------------------
Fix regression caused by #26620 and add test to demonstrate expected behaviour.

Before
----------------------------------------
`contribution.trxn_id` got changed to a single ID if there were multiple.

After
----------------------------------------
`contribution.trxn_id` are a comma-separated list.

Technical Details
----------------------------------------

Comments
----------------------------------------
@eileenmcnaughton

### CODE REVIEW

File: tests/phpunit/api/v3/PaymentTest.php [hunk 1]
```diff
@@ -882,6 +882,75 @@ public function testCreatePaymentPayLater(): void {
...
+    $this->assertEquals(100, $eft['values'][$eft['id']]['amount']);
+    // Check contribution for completed status
+    $contribution = $this->callAPISuccess('contribution', 'get', ['id' => $contribution['id']]);
+    $this->assertEquals('Completed', $contribution['values'][$contribution['id']]['contribution_status']);
+    $this->assertEquals('trxn_1,trxn_2', $contribution['values'][$contribution['id']]['trxn_id']);
+    $this->assertEquals(100.00, $contribution['values'][$contribution['id']]['total_amount']);
+    $this->callAPISuccess('Contribution', 'Delete', [
+      'id' => $contribution['id'],
+    ]);
+    $this->validateAllPayments();
```
Comment (eileenmcnaughton) [inline tests/phpunit/api/v3/PaymentTest.php:951]: Not a blocker but this validate isn't needed cos it is done by the parent before tear-down

File: CRM/Financial/BAO/Payment.php [hunk 2]
```diff
@@ -188,7 +188,6 @@ public static function create(array $params): CRM_Financial_DAO_FinancialTrxn {
           'is_post_payment_create' => TRUE,
           'is_email_receipt' => $params['is_send_contribution_notification'],
           'trxn_date' => $params['trxn_date'],
-          'trxn_id' => $params['trxn_id'] ?? NULL,
```
Comment (eileenmcnaughton) [inline CRM/Financial/BAO/Payment.php:191]: @mattwire so where does this get concatenated onto the existing value - `completeOrder` makes sense as a place to concatenate it on but I don't see it there

File: tests/phpunit/api/v3/PaymentTest.php [hunk 3]
```diff
@@ -882,6 +882,75 @@ public function testCreatePaymentPayLater(): void {
...
+      'total_amount' => 100,
+      'currency' => 'USD',
+      'contact_id' => $this->individualCreate(),
+      'financial_type_id' => 1,
+      'contribution_status_id' => 2,
+      'is_pay_later' => 1,
+      'trxn_id' => 'trxn_1'
+    ];
+    $contribution = $this->callAPISuccess('Contribution', 'create', $contributionParams);
+    //add payment for pay later transaction
```
Comment (eileenmcnaughton) [inline tests/phpunit/api/v3/PaymentTest.php:903]: Also non-blocking but the drupal standard (that we don't enforce) is a space & then capital letter & full stop at the end I think

File: CRM/Financial/BAO/Payment.php (hunk 2 above)
Comment (mattwire) [reply CRM/Financial/BAO/Payment.php:191]: It's concatenated in `CRM_Financial_BAO_Payment::updateRelatedContribution()`.

Comment (eileenmcnaughton) [reply CRM/Financial/BAO/Payment.php:191]: OK thanks - if jenkins is happy then that covers it!

**Comment by eileenmcnaughton** [comment]
@mattwire it's failing (the test fail that led me to make that change) & I think it is the `!empty` here https://github.com/civicrm/civicrm-core/blob/86b1c2f3331ea7e282995e92c608f36b8df2a76f/CRM/Financial/BAO/Payment.php#L253

**Comment by mattwire** [comment]
@eileenmcnaughton it's only authorize.net tests that are failing:
```
[CRM_Core_Payment_AuthorizeNetIPNTest.testIPNPaymentMembershipRecurSuccessNoLeakage]
[CRM_Core_Payment_AuthorizeNetIPNTest.testIPNPaymentMembershipRecurSuccessNoLeakageOnlineThenOffline]
```

If you fully revert #26620 those tests pass again so the test fails here are caused by changes in that PR. Not sure what to suggest, I'm just trying to get back to the previous core behaviour.

**Comment by eileenmcnaughton** [comment]
@mattwire did you look at the line I pointed to above - I suspect it is that - the issue is that when there is no trxn_id on the contribution & you call Payment.create it appears the trxn_id is not added to the contribution (so it's an Authorize.net test but not an Authorize.net specific issue)

**Comment by eileenmcnaughton** [comment]
Lets look at this next week - I don't want to create a tonne of work later to avoid 15 mins this week removing the empty check

**Comment by eileenmcnaughton** [comment]
This should pass when https://github.com/civicrm/civicrm-core/pull/26737 is merged (and can be merged at that point if it just has the original change of removing https://github.com/civicrm/civicrm-core/pull/26659/files#diff-74b281ac668a837edcbfabd5cb0d74b5647c92d24f63777ab964bc6376291e25L191 )

**Comment by mattwire** [comment]
test this please

**Comment by seamuslee001** [comment]
Jenkins re test this please

**Comment by mattwire** [comment]
@eileenmcnaughton Still failing on those authorizenet tests

**Comment by demeritcowboy** [comment]
Looks like a paypal network glitch not authnet? jenkins retest this please

**Comment by eileenmcnaughton** [comment]
Yeah passed this time

**Pull request merged**